### PR TITLE
feat: implement async queue with notify

### PR DESCRIPTION
**Pull request by wenym1**

### CODE REVIEW

File: foyer-common/src/queue.rs [hunk 1]
```diff
@@ -16,18 +16,16 @@ use std::{
...
-};
 
 #[derive(Debug)]
-pub struct AsyncQueue<T: Debug> {
-    tx: UnboundedSender<T>,
-    rx: Mutex<UnboundedReceiver<T>>,
-
+pub struct AsyncQueue<T> {
+    queue: Mutex<VecDeque<T>>,
     size: AtomicUsize,
```
Comment (MrCroxx) [inline foyer-common/src/queue.rs:27]: If `Mutex` is used on `VecQueue` for both `acqiure` and `release` method, `size: AtomicUsize` can be removed and use `RwLock` on `VecQueue` then use shared lock for `len`

Comment (wenym1) [reply foyer-common/src/queue.rs]: fixed

**Review by MrCroxx — APPROVED** [review]
LGTM!

No performance regression after bench.

```
> rm -rf /data/foyer && cargo build --release && RUST_BACKTRACE=1 RUST_LOG=warn ./target/release/foyer-storage-bench --dir /data/foyer --capacity 102400 --region-size 256 --lookup-range 10000000 --time 60 --flushers 8 --reclaimers 8 --entry-size 65536 2>&1 | tee ./target/bench.log

Total:
disk total iops: 141994.7
disk total throughput: 2.1 GiB/s
disk read iops: 41100.5
disk read throughput: 545.2 MiB/s
disk write iops: 100894.1
disk write throughput: 1.5 GiB/s
insert iops: 23929.8/s
insert throughput: 1.5 GiB/s
insert lat p50: 243us
insert lat p90: 279us
insert lat p99: 467us
get iops: 10111.1/s
get miss: 0.19%
get hit lat p50: 1143us
get hit lat p90: 3487us
get hit lat p99: 5951us
get miss lat p50: 3us
get miss lat p90: 3us
get miss lat p99: 4us
```

**Comment by MrCroxx** [comment]
@wenym1  Thanks for contributing. Please fix problems from `make check` before merged. 🥰

**Pull request merged**